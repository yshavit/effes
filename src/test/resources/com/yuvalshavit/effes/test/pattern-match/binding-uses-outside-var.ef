foo (b: BoolWrapper | Void) -> Void:
  t = True
  return case b of
    BoolWrapper(x): print t
    Void: print False

main -> Void:
  foo BoolWrapper(True)
  return Void
