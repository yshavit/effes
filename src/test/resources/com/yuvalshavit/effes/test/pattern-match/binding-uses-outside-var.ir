def foo (b: BoolWrapper | Void) -> Void:
  AssignStatement
    var: t (pos=0)
    val:
      CtorInvoke
        type: True
  ReturnStatement
    CaseExpression
      case:
        VarExpression
          arg: b
          pos: 0
      of BoolWrapper:
        MethodInvoke
          method: debugPrint [built-in]
          arg0:
            VarExpression
              var: t
              pos: 0
      of Void:
        MethodInvoke
          method: debugPrint [built-in]
          arg0:
            CtorInvoke
              type: False
