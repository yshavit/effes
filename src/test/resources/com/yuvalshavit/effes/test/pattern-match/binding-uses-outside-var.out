Your argument was: True
>> Void
